ci-bench/src/cachegrind.rs

```diff
@@ -233,61 +233,55 @@ impl Sub for InstructionCounts {
 
 /// Returns the detailed instruction diff between the baseline and the candidate
 pub fn diff(baseline: &Path, candidate: &Path, scenario: &str) -> anyhow::Result<String> {
-    // The latest version of valgrind has deprecated cg_diff, which has been superseded by
-    // cg_annotate. Many systems are running older versions, though, so we are sticking with cg_diff
-    // for the time being.
-
-    let tmp_path = Path::new("ci-bench-tmp");
-    let tmp = File::create(tmp_path).context("cannot create temp file for cg_diff")?;
-
-    // cg_diff generates a diff between two cachegrind output files in a custom format that is not
-    // user-friendly
-    let cg_diff = Command::new("cg_diff")
-        // remove per-compilation uniqueness in symbols, eg
-        // _ZN9hashbrown3raw21RawTable$LT$T$C$A$GT$14reserve_rehash17hc60392f3f3eac4b2E.llvm.9716880419886440089 ->
-        // _ZN9hashbrown3raw21RawTable$LT$T$C$A$GT$14reserve_rehashE
-        .arg("--mod-funcname=s/17h[0-9a-f]+E\\.llvm\\.\\d+/E/")
+    // callgrind_annotate formats the callgrind output file, suitable for comparison with
+    // callgrind_differ
+    let callgrind_annotate_base = Command::new("callgrind_annotate")
         .arg(
             baseline
                 .join(CACHEGRIND_OUTPUT_SUBDIR)
                 .join(scenario),
         )
+        .arg("--auto=no")
```

Comment:
Maybe worth a comment here indicating why we want to turn off automatic annotation of source files that contributed to the event count threshold?